### DISCUSSION TITLE
Dúvida sobre rotas

### DISCUSSION THREAD

**Discussion by Syscomp-Software** [OpenSourceCommunityBrasil/REST-DataWare#151] [Q&A] [answered]
Olá, Estou tentando separar as rotas em módulos separados, não sei se estou fazendo da maneira correta ou se é possível.
Estou fazendo como no código abaixo, apenas a rota de usuário responde, a rota de produto retorna que nenhum evento foi encontrado.
Os 2 módulos contem um RESTDWServerEvents com 1 "Api" cada.
A chamada que estou tentando ralizar são: 
protocolo://endereco:porta/api/v1/produto
protocolo://endereco:porta/api/v1/usuario

Versão 2.0.+
```
RESTDWIdServicePooler.AddDataRoute('api/v1/', TUsuarioController);
RESTDWIdServicePooler.AddDataRoute('api/v1/', `TProdutoController);
```

**Comment by mobius1qwe** [comment, marked as answer]
Sim, é através do AddDataRoute mesmo, porém ele não aceita subníveis, precisa informar a rota seguindo o padrão descrito [aqui](https://github.com/OpenSourceCommunityBrasil/REST-DataWare/wiki/URL-Flex%C3%ADvel)
no caso do seu exemplo, só existe uma rota que é a `/api/` e dentro dessa rota você tem endpoints com BaseURL diferentes: `/v1/produto`, `/v1/usuario` daí não teriam duas rotas, somente 1 por causa do padrão de nomenclatura

Seriam rotas diferentes se você fizesse:

`AddDataRoute('sisconf', TSISConfServerMethodDataModule);`
```
protocolo://endereco:porta/sisconf/baseurl
```
`AddDataRoute('resis', TReSISServerMethodDataModule);`
```
protocolo://endereco:porta/resis/baseurl
```
